liwords-ui/src/store/socket_handlers.ts

```diff
@@ -458,7 +460,14 @@ export const useOnSocketMsg = () => {
 
             console.log('got presence entry', pe.toObject());
 
-            // TODO.
+            setFriends({
+              ...friends,
+              [pe.toObject().userId]: {
```

Comment:
(performance) this can be `[pe.getUserId()]:`. `pe.toObject()` would first allocate a new instance with all key-value pairs, and then we only need one field.

Reply:
I can't get the typecasting to get past the compiler since PE.toObject has extra fields (deleting, anon). Leaving it. Feel free to patch later?